src/core/handlers/RequestHandler.ts

```diff
@@ -149,32 +147,10 @@ export abstract class RequestHandler<
    * Parse the intercepted request to extract additional information from it.
    * Parsed result is then exposed to other methods of this request handler.
    */
-  async parse(_args: {
-    request: Request
-    resolutionContext?: ResponseResolutionContext
-  }): Promise<ParsedResult> {
+  async parse(_args: { request: Request }): Promise<ParsedResult> {
     return {} as ParsedResult
   }
 
-  /**
-   * Test if this handler matches the given request.
-   */
-  public async test(args: {
```

Comment:
We only called `test` in tests.  Not sure if we need these anymore, but maybe this is only public to assert some behavior works?

If so we can extend the documentation, and maybe make this `private` but ignore that privacy in tests

Reply:
We could also change the signature here and call it directly in `run`

```ts
public async test(args: {
    request: Request
  }): Promise<[boolean, parsedResult]> {
    const parsedResult = await this.parse({
      request: args.request,
      resolutionContext: args.resolutionContext,
    })

    return [
      this.predicate({
          request: args.request,
          parsedResult, 
      }),
      parsedResult
   ]
  }
```

Reply:
Although we don't call this anywhere, the `.test()` method on the request handler is still a part of its public API. This becomes useful for custom request handlers as the developer can _test_ the handler against any request at any time. 

This is much more convenient than having to do parse + predicate manually (which is precisely what `.test()` abstracts).

I'd leave this method be, it's not a mistake. 

The action point here is to document the `.test()` method and give an example of usage in the [docs](https://mswjs.io/docs/api/request-handler).